### PR TITLE
Add Scout and fix some sentry issues

### PR DESCRIPTION
**Pull request by jpslav**
Add Scout and fix some sentry issues

Scout reporting:

image

Sentry reporting:

image

### CODE REVIEW

File: lib/rescue_from_unless_local.rb [hunk 1]
```diff
@@ -35,7 +37,12 @@ def rescue_from_unless_local(*klasses, with: nil, &block)
             end
           end
 
-        raise exception if Rails.application.config.consider_all_requests_local
+        if Rails.application.config.consider_all_requests_local
```
Comment (steveburkett) [inline lib/rescue_from_unless_local.rb:40]: we should prob. have a unit test for this file that makes sure the right messages are sent out (to Raven) if sentry is turned on.

File: lib/tasks/install_secrets.rake [hunk 2]
```diff
@@ -45,26 +45,40 @@ task :install_secrets, [] do
     end
   end
 
-  database_secrets = secrets.delete('database')
+  database_secrets = secrets.delete(:database)
+  write_yaml_file("config/database.yml", {
```
Comment (steveburkett) [inline lib/tasks/install_secrets.rake:49]: seems like this sort of thing should go into a gem called "openstax-rails" or something...it's something all rails apps at openstax need to do for boilerplate eh?

File: lib/tasks/install_secrets.rake [hunk 3]
```diff
@@ -45,26 +45,40 @@ task :install_secrets, [] do
...
+    production: {
+      database: database_secrets[:name],
+      host: database_secrets[:url],
+      port: database_secrets[:port],
+      username: database_secrets[:username],
+      password: database_secrets[:password],
+      adapter: "postgresql",
+      encoding: "unicode",
+      pool: '<%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>'
```
Comment (steveburkett) [inline lib/tasks/install_secrets.rake]: does 5 match what the needs of all puma's threads and workers? (curious if 5 has been reasoned out)

File: lib/tasks/install_secrets.rake [hunk 4]
```diff
@@ -45,26 +45,40 @@ task :install_secrets, [] do
...
+  })
+end
 
-  File.open(File.expand_path("config/secrets.yml"), "w") do |file|
-    # write the secrets hash as yaml, getting rid of the "---\n" at the front
-    file.write({'production' => secrets}.to_yaml[4..-1])
+def write_yaml_file(filename, hash)
+  File.open(File.expand_path(filename), "w") do |file|
+    # write the hash as yaml, getting rid of the "---\n" at the front
+    file.write(hash.deep_stringify_keys.to_yaml[4..-1])
```
Comment (steveburkett) [inline lib/tasks/install_secrets.rake:81]: would something like `yaml.gsub("---\n", '')` be better than these 2 lines?  it's one line and self documenting?

**Review by steveburkett — APPROVED** [review]

File: lib/tasks/install_secrets.rake (hunk 2 above)
Comment (jpslav) [reply lib/tasks/install_secrets.rake:49]: I would conceivable put the `write_yaml_file` in a gem, but not the specific lines here

File: lib/tasks/install_secrets.rake (hunk 3 above)
Comment (jpslav) [reply lib/tasks/install_secrets.rake]: good question.  I have no idea.  can I give one to you on the backburner?

File: lib/tasks/install_secrets.rake (hunk 4 above)
Comment (jpslav) [reply lib/tasks/install_secrets.rake:81]: good point.  we should do that once we move it somewhere and have a spec for it.

File: lib/tasks/install_secrets.rake (hunk 3 above)
Comment (jpslav) [reply lib/tasks/install_secrets.rake]: >If you are using the Puma web server we recommend setting the pool value to equal ENV['RAILS_MAX_THREADS']

https://devcenter.heroku.com/articles/concurrency-and-database-connections
so it should be 10.

File: lib/rescue_from_unless_local.rb (hunk 1 above)
Comment (jpslav) [reply lib/rescue_from_unless_local.rb:40]: agreed, but deferring to moving this into a gem (https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/673)

File: lib/tasks/install_secrets.rake (hunk 4 above)
Comment (jpslav) [reply lib/tasks/install_secrets.rake:81]: delayed to https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/673

**Pull request merged**